src/Persistence.LinnApps/Repositories/StorageLocationRepository.cs

```diff
@@ -20,22 +19,19 @@ public StorageLocationRepository(ServiceDbContext serviceDbContext) : base(servi
             this.serviceDbContext = serviceDbContext;
         }
 
-        public override async Task<IList<StorageLocation>> FindAllAsync()
+        public override IQueryable<StorageLocation> FindAll()
         {
-            return await this.serviceDbContext.StorageLocations
+            return this.serviceDbContext.StorageLocations
                        .Where(l => l.DateInvalid == null)
-                       .OrderBy(l => l.LocationCode)
-                       .ToListAsync();
+                       .OrderBy(l => l.LocationCode);
         }
 
-        public override async Task<IList<StorageLocation>> FilterByAsync(
-            Expression<Func<StorageLocation, bool>> filterExpression,
-            Expression<Func<StorageLocation, object>> orderByExpression = null)
+        public override IQueryable<StorageLocation> FilterBy(
+            Expression<Func<StorageLocation, bool>> filterExpression)
         {
-            return await this.serviceDbContext.StorageLocations
+            return this.serviceDbContext.StorageLocations
                        .Where(filterExpression)
-                       .OrderBy(l => l.LocationCode)
-                       .ToListAsync();
+                       .OrderBy(l => l.LocationCode);
```

Comment:
Probably better to pass the order expressions from the consuming code as opposed to hardcoding them here, now that the IRepository supports orderByExpressions